File: Templates/SpiderEye.Templates/templates/MyApp.FSharp/MyApp.Core/ProgramBase.fs
module MyApp.Core
open SpiderEye

let mainBase() =
    use window = new Window()
    window.Title <- "MyApp"
    window.Icon <- AppIcon.FromFile("icon", ".")

    Application.ContentProvider <- EmbeddedContentProvider("App")
    Application.Run(window, "index.html")
    0

File: Examples/Simple/App.Core/ProgramBase.cs
namespace SpiderEye.Example.Simple.Core
{
    public abstract class ProgramBase
    {
        protected static void Run()
        {
            // this creates a new window with default values
            using (var window = new Window())
            {
                // this relates to the path defined in the .csproj file
                Application.ContentProvider = new EmbeddedContentProvider("App");

                // runs the application and opens the window with the given page loaded
                Application.Run(window, "index.html");
            }
        }
    }
}
